### DISCUSSION TITLE
Error (Assertion `args[3]->IsInt32()' failed) using a simple create

### DISCUSSION THREAD

**Discussion by carlitorweb** [prisma/prisma#6831] [Q&A] [answered]
In a Express + Nodejs api

**client.ts (to instance the prisma object)**
```js
import { PrismaClient } from '@prisma/client';

// add prisma to the NodeJS global type
interface CustomNodeJsGlobal extends NodeJS.Global {
    prisma: PrismaClient;
}

// Prevent multiple instances of Prisma Client in development
declare const global: CustomNodeJsGlobal;

const prisma = global.prisma || new PrismaClient();

if (process.env.NODE_ENV === 'development') global.prisma = prisma;

export default prisma;
```

**post.ts (where I get the error)**
```js
// createPost Function
const createPost = async (req: FeedRequest, res: Response): Promise<void | Response> => {
    const errors = validationResult(req);
    if (!errors.isEmpty()) {
        return res.status(422).json({
            message: 'Validation fail, entered data is incorrect',
            errors: errors.array(),
        });
    }

    const { title, content, creator } = req.body;

    const post = await prisma.posts.create({
        data: {
            title,
            content,
            creator,
        },
    });

    res.status(201).json({
        message: 'Post created successfully!',
        post: post,
    });
};
```
### Error when I try to use the endpoint to create a post
```sh
node/v16.0.0/bin/node[14161]: ../src/node_http_parser.cc:567:static void node::{anonymous}::Parser::Initialize(const v8::FunctionCallbackInfo<v8::Value>&): Assertion `args[3]->IsInt32()' failed.
1: 0xb12b00 node::Abort() [/versions/node/v16.0.0/bin/node]
2: 0xb12b7e  [/versions/node/v16.0.0/bin/node]
3: 0xb2a282  [/versions/node/v16.0.0/bin/node]
4: 0xd5f70b  [/versions/node/v16.0.0/bin/node]
5: 0xd60bac  [/versions/node/v16.0.0/bin/node]
6: 0xd61226 v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) [/versions/node/v16.0.0/bin/node]
[Node]  7: 0x160c579  [/versions/node/v16.0.0/bin/node]
```

If I remove the `const post = await prisma.posts.create...` part, all work okay. I really not sure what is happening here

**Comment by carlitorweb** [comment, marked as answer]
The current prisma version (2.21.2) is not compatible yet with the current last version (16) of nodejs. Downgrade temporary to the last LTS of nodejs fix the problem.

**Reply by janpio** [reply]
2.22.0 release next week will support Node 16 as well 👍

**Comment by shirshak55** [comment]
I solved it by using latest dev version

yarn add @prisma/client@dev prisma@dev

For me it is:
```
  "@prisma/client": "^2.22.0-dev.48",
  ```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
